packages/desktop-client/src/components/manager/ConfigServer.tsx

```diff
@@ -101,10 +101,23 @@ export function ConfigServer() {
 
     if (selfSignedCertificateLocation) {
       setServerSelfSignedCert(selfSignedCertificateLocation[0]);
-      globalThis.window.Actual.relaunch(); // relaunch to use the certificate
     }
   }
 
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    // When self signed cert has changed, restart the server
+    if (isMounted.current) {
+      setLoading(true);
+      globalThis.window.Actual.restartElectronServer();
+      setError(null);
+      setLoading(false);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏the two `setLoading` operations will happen in the same render cycle so there's not much point in having them.